crates/api/src/types.rs

```diff
@@ -433,29 +496,38 @@ impl ImportType {
 /// [`Module::exports`](crate::Module::exports) accessor and describes what
 /// names are exported from a wasm module and the type of the item that is
 /// exported.
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
-pub struct ExportType {
-    name: String,
-    ty: ExternType,
+#[derive(Clone, Hash, Eq, PartialEq)]
+pub struct ExportType<'module> {
+    /// The name of the export.
+    name: &'module str,
+
+    /// The type of the export.
+    ty: EntityType<'module>,
 }
 
-impl ExportType {
+impl<'module> ExportType<'module> {
     /// Creates a new export which is exported with the given `name` and has the
     /// given `ty`.
-    pub fn new(name: &str, ty: ExternType) -> ExportType {
-        ExportType {
-            name: name.to_string(),
-            ty,
-        }
+    pub(crate) fn new(name: &'module str, ty: EntityType<'module>) -> ExportType<'module> {
+        ExportType { name, ty }
     }
 
     /// Returns the name by which this export is known by.
-    pub fn name(&self) -> &str {
-        &self.name
+    pub fn name(&'module self) -> &'module str {
```

Comment:
I think the `'module self` can just be `self` here (keeping the `&'module str` ret)

Reply:
Done